### PR TITLE
Unexcused absences Overview in Tutorium detail page

### PR DESCRIPTION
**Pull request by EricZimmer1**
Added a table containing all unexcused absences for all students of a Tutorium.
An accordion was used to group both tables together.

### CODE REVIEW

**Review by jonaslanglotz — COMMENTED** [review]
Could we maybe include some links or other function to make this page more helpful? Maybe we can include a link to the student page in each absence?

**Comment by EricZimmer1** [comment]
Sounds like a good idea. what should be clickable? The whole table entry or just the name?

**Comment by jonaslanglotz** [comment]
I was thinking just the thing we're linking to, so probably the students name.

**Comment by bucherfa** [comment]
@EricZimmer1 maybe also an 'entschuldigen' button, you could just create a single hour excuse once the button is clicked. I actually expected this feature also appear on the student overview page. Could you add that too?

**Comment by EricZimmer1** [comment]
@bucherfa the entry already is quite big. Don’t know how well it would fit. Any idea where to put it?

**Comment by bucherfa** [comment]
@EricZimmer1 to shink down the size of a row: replace the text "Eingereicht von" with an icon. And generally use the small text size from chakra ui

**Comment by EricZimmer1** [comment]
image
Still got these two errors, however server is running fine.

File: web/src/pages/student/[id].tsx [hunk 1]
```diff
@@ -190,20 +229,20 @@ const Student: React.FC<Props> = ({ self }) => {
...
-                                  <Text>Eingereicht von <span style={{ fontWeight: 'bold' }}>{row.cells[1].render('Cell')}</span></Text>
+                                  <Text fontSize="sm"fontWeight="bold">{row.cells[0].render('Cell')}. Stunde</Text>
+                                  <Text fontSize="sm"><Icon as={FaUser} mr={2} mb={1} />{row.cells[1].render('Cell')}</Text>
                                 </Flex>
                                 <Spacer />
                                 <Flex flexDirection="column">
-                                  {row.cells[3].value ? (<Tag mb={2} bgColor="blue.400" color="white">Klausur</Tag>) : (<></>)}
-                                  {row.cells[3].value ? (<Tag colorScheme="green" variant="solid">Entschuldigt</Tag>) : (<></>)}
+                                  {row.cells[2].value ? (<Tag fontSize="sm" mb={2} bgColor="blue.400" color="white">Klausur</Tag>) : (<></>)}
+                                  {row.cells[3].value ? (<Tag fontSize="sm" colorScheme="green" variant="solid">Entschuldigt</Tag>) : (<Tag fontSize="sm" mb={2} colorScheme="red">Unentschuldigt</Tag>)}
```
Comment (bucherfa) [inline web/src/pages/student/[id].tsx]: tags have a size prop https://chakra-ui.com/docs/data-display/tag#props

might be better than changing the font size

File: web/src/pages/tutorium/[id].tsx [hunk 2]
```diff
@@ -47,13 +31,33 @@ const StudentsOfTutoriumPage: React.FC<Props> = ({ self }) => {
...
+    e.absences.forEach(absence => {
+      if (absence.excused === false) {
+        absences.push(absence)
+      }
+    })
+  })
+
+  const dates = Array.from(new Set(absences.map(absence => absence.date))).sort().reverse()
+
+  console.log(absences)
```
Comment (bucherfa) [inline web/src/pages/tutorium/[id].tsx]: log can be removed

**Review by bucherfa — DISMISSED** [review]

File: web/src/pages/student/[id].tsx (hunk 1 above)
Comment (EricZimmer1) [reply web/src/pages/student/[id].tsx]: changed that

File: web/src/pages/tutorium/[id].tsx (hunk 2 above)
Comment (EricZimmer1) [reply web/src/pages/tutorium/[id].tsx]: removed

**Review by bucherfa — APPROVED** [review]

**Review by jonaslanglotz — APPROVED** [review]

**Pull request merged**